src/main/java/embl/ebi/variation/commons/models/metadata/Study.java

```diff
@@ -183,14 +184,21 @@ public void setUris(Set<URI> uris) {
         return Collections.unmodifiableSet(fileGenerators);
     }
 
+    @PreRemove
```

Comment:
Right, this code is called when an study is going to be removed from the database. We need it if we want to delete the study from the repository without having to "unset" manually the file generators from the study. 

If we execute the testDeleteStudyButNotAnalysys method removing the "PreRemove" annotation from the Study class, we will get a DataIntegrityViolationException because we will violate  fk_study_id from file_generator. To avoid that, we can add the PreRemove annotation to unset all file generators, or "remove" the file generators from the study before deleting it:

        savedStudy.removeFileGenerator(analysis1);
        savedStudy.removeFileGenerator(analysis2);
        repository.delete(savedStudy);